src/components/ui/NavBar.tsx

```diff
@@ -0,0 +1,71 @@
+import { type ReactNode } from "react";
+import { type IconType } from "react-icons/lib";
+import { TbScubaMask, TbHistory, TbAward } from "react-icons/tb";
+
+type NavItemProps = {
+  title: string;
+  Icon: IconType;
+  pageNo: number;
+  pageHandler: (pageNo: number) => void;
+};
+
+type NavBarProps = {
+  children: ReactNode;
+  page: number;
+  pageHandler: (pageNo: number) => void;
+};
+
+const NavItem = ({ title, Icon, pageNo, pageHandler }: NavItemProps) => {
+  const onChangePage = () => {
+    pageHandler(pageNo);
+  };
+  return (
+    <div className="flex w-1/3 flex-col justify-center" onClick={onChangePage}>
+      <Icon size={36} className="mx-auto text-secondary" />
+      <p className="text-center text-sm font-semibold text-secondary">
+        {title}
+      </p>
+    </div>
+  );
+};
+
+const NavBar = ({ children, page, pageHandler }: NavBarProps) => {
```

Comment:
don't define children as `ReactNode`, instead use `PropsWithChildren`

```suggestion
const NavBar: FC<PropsWithChildren<NavBarProps>> = ({ children, page, pageHandler }) => {
```